### PR TITLE
Adds SimpleMandateElement, require it to be displayed 

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
1. Creates a `SimpleMandateElement` that requires its view to have been displayed before producing a valid payment option.
2. Preserves "view was displayed" state in IntentConfirmParams.

## Motivation
Make `PaymentSheet.FlowController.update` flow preserve customer input when the customer is using a payment method that can require a mandate, like Paypal.  Specifically, handle edge cases around the payment method being updated to include or not include a mandate e.g. going from payment -> setup mode.  See the unit tests for details.

## Testing
See unit test

## Changelog
No user facing changes

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/SimpleMandateElement.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if mandateTextView.viewDidAppear {
+            params.didDisplayMandate = true
+        }
+        if customerAlreadySawMandate || mandateTextView.viewDidAppear {
+            // the customer must have seen the mandate for this to be valid
+            return params
+        } else {
+            return nil
+        }
+    }
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/SimpleMandateElement.swift:22]: This is the core logic fyi @porter-stripe @tillh-stripe @jameswoo-stripe

**Review by porter-stripe — DISMISSED** [review]
cool solution!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/SimpleMandateElement.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//  StripePaymentSheet
+//
+//  Created by Yuki Tokuhiro on 3/26/23.
+//
+
+@_spi(STP) import StripeUICore
+import UIKit
+
+class SimpleMandateElement: PaymentMethodElement {
+    func updateParams(params: IntentConfirmParams) -> IntentConfirmParams? {
```
Comment (porter-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/SimpleMandateElement.swift:12]: Nit:  Could we maybe add a little more comments around this logic and what it means for these params to be nil.

Comment (yuki-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/SimpleMandateElement.swift:12]: Thanks for the suggestion.  As I wrote it up, I realized we should set `didDisplayMandate = true` in the case where the customer already saw the mandate too.

**Review by porter-stripe — APPROVED** [review]
